crates/ruff_notebook/src/cell.rs

```diff
@@ -170,7 +170,50 @@ impl Cell {
         }
 
         // Detect cell magics (which operate on multiple lines).
-        lines.any(|line| line.trim_start().starts_with("%%"))
+        lines.any(|line| {
+            line.split_whitespace().next().is_some_and(|first| {
+                if first.len() < 2 {
+                    return false;
+                }
+                let (token, command) = first.split_at(2);
+                // These cell magics are special in that the lines following them are valid
+                // Python code and the variables defined in that scope are available to the
+                // rest of the notebook.
+                //
+                // For example:
+                //
+                // Cell 1:
+                // ```python
+                // x = 1
+                // ```
+                //
+                // Cell 2:
+                // ```python
+                // %%time
+                // y = x
```

Comment:
So how does the parser handle this case?

Reply:
The parser would parse the first line `%%time` as a cell magic statement and the second line as an assignment statement.

This is technically a bug in the parser as both lines should be a single node instead but as we always ignored cell magic, it never came up.

Reply:
Where does the cell magic end? Does it just include the `%%time` regardless of what follows? Like how would the parser deal with:

```python
%%time x = 1
y = x
```

Reply:
Each statement is a single node, so `%%time x = 1` would a single node and `y = x` would be an assignment statement. This code would raise undefined name for `x`. This is intentional as that requires support for inline parsing which might require more thinking.

Reply:
I'm mostly wondering how the parser knows where the magic statement ends.

Reply:
Oh, it's the newline. The lexer consumes everything till the newline and considers this as part of the command.

Reply:
Yeah, but that is ignored. So,
```python
%%time \
	x = 1
y = x
```

will parse in same way as your example.